File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float GetPositiveFloat(void)
{
 float n;
    do
    {
        printf("O hai, ow much is owed ?");
       n = GetFloat();
    }
   while (n < 0);
    return n;
}

int main(void)
{
    int coins = 0;
  
    
    printf("How much change is owed? \n");
    float c = GetPositiveFloat();
    int change = round(c * 100);
    

    while(change % 25 >= 0 && change >= 25 && change > 0){
        change -= 25;
        coins++;
    }
    
     while(change % 10 >= 0 && change >= 10 && change > 0){
        change -= 10;
        coins++;
    }
    
     while(change % 5 >= 0 && change >= 5 && change > 0){
        change -= 5;
        coins++;
    }
    
    while(change % 1 == 0 && change >= 1 && change > 0){
        change -= 1;
        coins++;
    }
    

printf("%i\n",coins);
    
}

File: pset6/greedy.py
import cs50

change = 0
while True:
    print("O hai, how much is owed ?", end="")
    change = cs50.get_float()
    if change >= 0:
        break;


coins = 0
cents = int(round(change * 100))

coins += cents // 25
cents = cents % 25

coins += cents // 10
cents = cents % 10

coins += cents // 5
cents = cents % 5

coins += cents // 1
cents = cents % 1
    
print(coins)
    


    
    